src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/Transform.scala

```diff
@@ -166,4 +153,32 @@ object Transform {
   private val secondPassTransformMap: TransformMap =
     // Overwrite collector-set nuid with tracker-set tnuid
     Map(("tnuid", (ME.toTsvSafe, "network_userid")))
+
+  def setCollectorTstamp(event: EnrichedEvent, timestamp: Option[DateTime]): Either[FailureDetails.EnrichmentFailure, Unit] =
+    EE.formatCollectorTstamp(timestamp).map { t =>
+      event.collector_tstamp = t
+      ().asRight
+    }
+
+  def setUseragent(
+    event: EnrichedEvent,
+    useragent: Option[String],
+    encoding: String
+  ): Either[FailureDetails.EnrichmentFailure, Unit] =
+    useragent match {
+      case Some(ua) =>
+        CU.decodeString(Charset.forName(encoding), ua)
+          .map { ua =>
+            event.useragent = ua
+            ()
```

Comment:
Is `()` required?

Reply:
It's not. I didn't take a look at these functions